index.js

```diff
@@ -1269,6 +1272,12 @@ class Hyperdrive extends Nanoresource {
   }
 }
 
+function HyperdriveCompat (...args) {
+  if (!(this instanceof HyperdriveCompat)) return new HyperdriveCompat(...args)
+  Hyperdrive.prototype._initialize.call(this, ...args)
```

Comment:
you need to call Nanoresource.call(this) before this line due to the inheritance

Reply:
Good catch, sorry I missed that. Fixed

Reply:
JS galore